ngraph/core/src/function.cpp

```diff
@@ -347,6 +384,37 @@ bool Function::visit_attributes(AttributeVisitor& visitor)
     return true;
 }
 
+void Function::add_sinks(const SinkVector& sinks)
+{
+    m_sinks.insert(m_sinks.end(), sinks.begin(), sinks.end());
+    validate_nodes_and_infer_types();
+}
+
+void Function::remove_sink(const std::shared_ptr<op::Sink>& sink)
+{
+    m_sinks.erase(std::remove_if(m_sinks.begin(),
+                                 m_sinks.end(),
+                                 [&sink](std::shared_ptr<op::Sink>& s) { return s == sink; }),
+                  m_sinks.end());
+    validate_nodes_and_infer_types();
+}
+
+void Function::add_results(const ResultVector& results)
+{
+    m_results.insert(m_results.end(), results.begin(), results.end());
+    validate_nodes_and_infer_types();
+}
+
+void Function::remove_result(const std::shared_ptr<op::Result>& result)
+{
+    m_results.erase(
+        std::remove_if(m_results.begin(),
+                       m_results.end(),
+                       [&result](std::shared_ptr<op::v0::Result>& r) { return r == result; }),
+        m_results.end());
+    validate_nodes_and_infer_types();
```

Comment:
Why do we need to re-validate a function after removing some results? It doesn't contribute new nodes into the graph.

Reply:
answered below